submission.go

```diff
@@ -53,3 +55,182 @@ func (h *Submission) String() string {
 	comments := fmt.Sprintf("%d comment%s", h.NumComments, plural)
 	return fmt.Sprintf("%d - %s (%s)", h.Score, h.Title, comments)
 }
+
+// SubredditSubmissions returns the submissions on the given subreddit.
+func (s Session) SubredditSubmissions(subreddit string) ([]*Submission, error) {
+	req := request{
+		url:       fmt.Sprintf("http://www.reddit.com/r/%s.json", subreddit),
+		useragent: s.useragent,
+	}
+	body, err := req.getResponse()
+	if err != nil {
+		return nil, err
+	}
+
+	type Response struct {
+		Data struct {
+			Children []struct {
+				Data *Submission
+			}
+		}
+	}
+
+	r := new(Response)
+	err = json.NewDecoder(body).Decode(r)
+	if err != nil {
+		return nil, err
+	}
+
+	submissions := make([]*Submission, len(r.Data.Children))
+	for i, child := range r.Data.Children {
+		submissions[i] = child.Data
+	}
+
+	return submissions, nil
+}
+
+// Returns top submmissions from a subreddit
+// Second argument must have value of one of "hour", "day", "week", "month", "year", "all"
+func (s Session) TopSubmissions(subreddit string, time string) ([]*Submission, error) {
+	times := []string{"hour", "day", "week", "month", "year", "all"}
```

Comment:
Maybe these should be their own enum type so people are left guessing what strings to use